### PR TITLE
[NEW] REST API endpoint `/directory`

### LINKED ISSUE #9961: [NEW] Directory REST APIs

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Add REST /directory endpoint.
Closes #9961.

### CODE REVIEW

File: tests/end-to-end/api/00-miscellaneous.js [hunk 1]
```diff
@@ -98,4 +98,93 @@ describe('miscellaneous', function() {
...
+					expect(res.body.result[0]).to.have.property('_id');
+					expect(res.body.result[0]).to.have.property('name');
+					expect(res.body.result[0]).to.have.property('usernames').and.to.be.an('array');
+					expect(res.body.result[0]).to.have.property('ts');
+				})
+				.end(done);
+		});
+	});
+
+	describe('/settings.oauth', () => {
```
[the remark this answers is not in the document]
Comment (sampaiodiego) [reply tests/end-to-end/api/00-miscellaneous.js]: oops.. looks it's duplicated of [this one](https://github.com/RocketChat/Rocket.Chat/pull/10299/files#diff-098e4a7532d065473fa77903fc8f3bb0R69).

@MarcosSpessatto can you please remove this test from here?

File: packages/rocketchat-api/server/v1/misc.js [hunk 2]
```diff
@@ -168,3 +168,26 @@ RocketChat.API.v1.addRoute('shield.svg', { authRequired: false }, {
...
+
+		const result = Meteor.runAsUser(this.userId, () => Meteor.call('browseChannels', {
+			text,
+			type,
+			sort: sortDirection,
+			page: offset,
+			limit: count
+		}));
+
+		if (result) {
```
Comment (sampaiodiego) [inline packages/rocketchat-api/server/v1/misc.js]: I think as a good practice, the unexpected condition should be treated specially and not the successful one.

so I will ask you to change this here, so the `if` will trigger the failure.

**Comment by graywolf336** [comment]
@rodrigok did you test this yourself? We were investigating an issue with the rest API not working on the deployed heroku instance.

**Comment by sampaiodiego** [comment]
@graywolf336 the last instance I tried on heroku was built based on `develop` branch instead of PR's code.. not sure it was the same for this one, just sharing

**Pull request merged**